pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/LegacyKitResampleRequest.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.dsm.service.adminoperation;
+
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+
+@Getter
+@AllArgsConstructor
+/**
+ * Request to resample a kit for a participant with a legacy short ID
+ * The request is for resampling a kit from Pepper Guid and Hruid to their old legacy ids
+ */
+public class LegacyKitResampleRequest {
+    String currentCollaboratorSampleId;
+    String newCollaboratorSampleId;
+    String newCollaboratorParticipantId;
+    String shortId;
+    String legacyShortId;
+
+    /**
+     * Verify that the request is valid,
+     * that all the fields are present,
+     * that the participant exists in the ES with the given short ID,
+     * that the given legacy short ID matches the one on file,
+     * that the kit request exists for the given collaborator sample ID to resmaple,
+     * and also that the new collaborator sample ID does not already exist
+     *
+     * @param ddpInstance DDP instance for the request
+     * @param kitRequestDao DAO for kit requests
+     */
+    public void verify(DDPInstance ddpInstance, KitRequestDao kitRequestDao) {
+        if (ddpInstance == null) {
+            throw new DsmInternalError("DDP instance not found");
+        }
+        checkNotEmptyRequestFields();
+        Map<String, Map<String, Object>> esParticipantData = ElasticSearchUtil.getSingleParticipantFromES(ddpInstance.getName(),
+                ddpInstance.getParticipantIndexES(), shortId);
+        if (esParticipantData.size() != 1) {
+            throw new DSMBadRequestException("Invalid participant short ID " + shortId);
+        }
+        Map<String, Object> ptp = esParticipantData.values().stream().findFirst().orElseThrow();
```

Comment:
Also, to avoid calling services twice for each request, you can consider doing this as you process the requests in the service.
This method cuts down on possible errors during processing vs catching them during the process and noting them as errors in the logs. I can see either approach as working, just wanted to mention it as an alternative.